### ISSUE TITLE
Prevent model from being created if there is any unknown attribute

### ISSUE THREAD

**Issue by Stackle** [solnic/virtus#337]
For example,
```ruby
class RegisterModel
    include Virtus.model(unknown: false)
    attribute :name, String
end

RegisterModel.new(name: "Ben Smith", mal_input: "---") #this should fail and raise error
```

**Comment by magicalbanana** [comment]
You can just override the setter and do the logic there.

**Comment by fr33z3** [comment]
I don't think that the behaviour when model raises exception when you are trying to get an instance of it is useful. Sounds like anti-pattern. Instead of that you should probably define some method named ```valid?``` to check if model is valid or not and then manually raise and exception.

**Comment by zhulik** [comment]
Sorry for necroposing, but this is default behavior in ActiveRecord and can really save you from typos in you code. I think optional mode with restricted unknown fields is a good idea.
